### PR TITLE
SDI-554 WIP for SENTENCE_ADJUSTMENT_UPSERTED event

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingSynchronisationService.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Service
+
+@Service
+class SentencingSynchronisationService(
+  private val sentencingMappingService: SentencingMappingService,
+  private val telemetryClient: TelemetryClient
+) {
+  suspend fun synchroniseSentenceAdjustmentCreateOrUpdate(event: SentenceAdjustmentUpsertedOffenderEvent) {
+    sentencingMappingService.findNomisSentencingAdjustmentMapping(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingSynchronisationService.kt:12]: TODO - More or less everything !!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/sentencing/SentencingMigrationIntTest.kt [hunk 2]
```diff
@@ -95,7 +95,7 @@ class SentencingMigrationIntTest : SqsIntegrationTestBase() {
       mappingApi.stubAllNomisSentencingAdjustmentsMappingNotFound()
       mappingApi.stubSentenceAdjustmentMappingCreate()
 
-      sentencingApi.stubCreateSentencingAdjustment()
+      sentencingApi.stubCreateSentencingAdjustmentForMigration()
       mappingApi.stubSentenceAdjustmentMappingByMigrationId(count = 86)
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/sentencing/SentencingMigrationIntTest.kt:99]: We expect 2 endpoint in Digital Wallet that more or less the same thing, one for migration for synchronisation

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/wiremock/MappingApiMockServer.kt [hunk 3]
```diff
@@ -554,15 +582,16 @@ class MappingApiMockServer : WireMockServer(WIREMOCK_PORT) {
   fun createSentenceAdjustmentMappingCount() =
     findAll(postRequestedFor(urlPathEqualTo("/mapping/sentencing/adjustments"))).count()
 
-  fun verifyCreateMappingSentenceAdjustmentIds(nomsSentenceAdjustmentIds: Array<Long>, times: Int = 1) = nomsSentenceAdjustmentIds.forEach {
-    verify(
-      times,
-      postRequestedFor(urlPathEqualTo("/mapping/sentencing/adjustments")).withRequestBody(
-        matchingJsonPath(
-          "adjustmentId",
-          equalTo("$it")
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/wiremock/MappingApiMockServer.kt:563]: just formatting

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingSynchronisationService.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          "sentenceSequence" to event.sentenceSeq.toString(),
+          "nomisAdjustmentId" to event.adjustmentId.toString(),
+          "adjustmentCategory" to "SENTENCE",
+          "adjustmentId" to it.adjustmentId,
+        ),
+        null
+      )
+    } ?: let {
+      telemetryClient.trackEvent(
+        "sentence-adjustment-created-synchronisation-success",
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingSynchronisationService.kt:30]: Should this have a different event name to above - fail?

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingSynchronisationService.kt:30]: one is created and one is updated

**Review by mikehalmamoj — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/sentencing/SentencingFromNomisIntTest.kt [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.integration.sendMessage
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.wiremock.MappingApiExtension.Companion.mappingApi
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.wiremock.NomisApiExtension.Companion.nomisApi
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.wiremock.SentencingApiExtension.Companion.sentencingApi
+import uk.gov.justice.hmpps.sqs.countAllMessagesOnQueue
+
+class SentencingFromNomisIntTest : SqsIntegrationTestBase() {
+
+  @SpyBean
+  private lateinit var telemetryClient: TelemetryClient
```
[the remark this answers is not in the document]
Comment (andymarke) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/sentencing/SentencingFromNomisIntTest.kt:29]: Ah, OK - next PR - thanks

**Pull request merged**